### PR TITLE
Review ShubhamD

### PR DESCRIPTION
**Pull request by deka1105**
Please review!

### CODE REVIEW

File: HW03_Shubham_Dekatey_dictionary.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        try:
+            with open("filterWords.txt", "w") as outfile:
+                outfile.write("\n".join(str(item) for item in lenWordList))
+            log.writeLog("filterWords.txt Created")
+            return len(lenWordList)
+        except:
+            print("filterWords.txt creation error")
+            log.writeLog("filterWords.txt creation error")
+
+    def getWordfromFile():
```
Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_dictionary.py:86]: Methods must have a first parameter, usually called 'self' so its missing here.

Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_dictionary.py:86]: Methods must have a first parameter, usually called 'self' so its missing here.

File: HW03_Shubham_Dekatey_ui.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+# word = ""
+# wordLength = 0
+# attempts = []
+
+class UiX:
+
+
+    globalAttempts = []
+
+    def initWordle():
```
Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_ui.py:13]: Usually need to passing parameters with the methods called 'self' which is missing over here as well.

Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_ui.py:13]: Usually need to passing parameters with the methods called 'self' which is missing over here as well.

File: HW03_Shubham_Dekatey_wordle.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            if(resultString == ""):
+                winCount += 1
+        except:
+            print("Game Wordle.checker Error")
+    
+
+    def init(self):
+        #global correctWord
+        try:
+            gameCount += 1
```
Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_wordle.py:72]: it's throwing me an error called unresolved error reference called 'gameCount'.here and there.

Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_wordle.py:72]: it's throwing me an error called unresolved error reference called 'gameCount'.here and there.

File: TesterIO.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        import pathlib as pl
+        path = pl.Path("letterFrequency.csv")
+        self.assertEquals((str(path), path.is_file()), (str(path), True))
+        
+    def check_file_rank(self) -> None:
+        """Check if the Rank CSV file is created"""
+        import pathlib as pl
+        path = pl.Path("wordRank.csv")
+        self.assertEquals((str(path), path.is_file()), (str(path), True))
+
```
Comment (jaiminbd13) [inline TesterIO.py:70]: looks great..! all the functions defined cleanly nd nicely.

Comment (jaiminbd13) [inline TesterIO.py:70]: looks great..! all the functions defined cleanly nd nicely.

File: Wordle_One_Word.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    while(valid_inp == 0):
+        word_inp = input("Enter your value (Only first 5 characters will be considered): ")
+        word_inp = word_inp[:5]
+
+        #Check input length, should be greater than length of word
+        if(len(word_inp) >= len(word)):
+            # Check if the first 5 characters are only alphabets
+            if re.match("^[a-z, A-Z]*$", word_inp):
+                print("Your word:", word_inp)
+
```
Comment (jaiminbd13) [inline Wordle_One_Word.py:19]: looks so standardized..! great work on pseudocodes as well, nice readability.

Comment (jaiminbd13) [inline Wordle_One_Word.py:19]: looks so standardized..! great work on pseudocodes as well, nice readability.

File: freqRank.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+from logger import LogX
+
+
+listWords = []
+freqDict = {}
+occDict = {}
+allKeys = []
+freqTuple = {}
+freqTup = {}
+
```
Comment (jaiminbd13) [inline freqRank.py:14]: the best part I have noticed in your code the variables defined so greatly and relevant with their names, Great work done...!!

Comment (jaiminbd13) [inline freqRank.py:14]: the best part I have noticed in your code the variables defined so greatly and relevant with their names, Great work done...!!

File: logger.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+
+class LogX:
+    def writeLog(inputString):
+        try:
+            text_file = open("log.txt", "w")
+            n = text_file.write(inputString)
```
Comment (jaiminbd13) [inline logger.py:6]: Excellent work...!

Comment (jaiminbd13) [inline logger.py:6]: Excellent work...!

File: solver.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            print(chars)
+            df.loc[len(df)] = chars
+
+        newdf = pd.DataFrame(columns=['1','2','3','4','5'])
+        for i in rIndexList:
+            charX = rCList[i]
+            newdf = df[df[str(i)] == charX]
+        
+        lsX = newdf.index.values.tolist()
+        compatibleWords = self.listToWord(lsX, wordList)
```
Comment (jaiminbd13) [inline solver.py:59]: showing me an error called Unresolved Error called 'self'.

Comment (jaiminbd13) [inline solver.py:59]: showing me an error called Unresolved Error called 'self'.

File: solver.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
+from glob import glob
+from unittest import result
+from HW03_Shubham_Dekatey_ui import UiX as ui
+from HW03_Shubham_Dekatey_wordle import WordleX as wordle
+from HW03_Shubham_Dekatey_dictionary import DictionaryX as dictionary
+from tipsHelper import TipsHX
+import pandas as pd
```
Comment (jaiminbd13) [inline solver.py:7]: please check the module here which is importing. showing me no modules rest of the code looks great..!

File: tipsHelper.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    csvWordList = []
+    # goodWords = []
+    # badWords = []
+    result = []
+    def __init__(self):
+        #read csv
+        global csvWordList
+        try:
+            data = pd.read_csv("wordRank.csv")
+            csvWordList = data['word'].tolist()
```
Comment (jaiminbd13) [inline tipsHelper.py:18]: done a great work here...! all the syntax looks so clean nd nicely managed.

Comment (jaiminbd13) [inline tipsHelper.py:18]: done a great work here...! all the syntax looks so clean nd nicely managed.

File: tipsHelper.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
+from glob import glob
+from unittest import result
+from HW03_Shubham_Dekatey_ui import UiX as ui
+from HW03_Shubham_Dekatey_wordle import WordleX as wordle
+from HW03_Shubham_Dekatey_dictionary import DictionaryX as dictionary
+import pandas as pd
```
Comment (jaiminbd13) [inline tipsHelper.py:6]: need to check the module 'pandas' which imported here.

Comment (jaiminbd13) [inline tipsHelper.py:6]: need to check the module 'pandas' which imported here.

**Comment by jaiminbd13** [comment]
over all, one of the best codes I have ever seen and checked. Greatly done each nd every variable functions nd syntax are looks so clean and easy to readable also very specific professionally named. few of things which I guess needs to be changed or corrected like need to check the import modules in some files and missing the parameters with the methods. learned a lot as well.

**Comment by jaiminbd13** [comment]
over all, one of the best codes I have ever seen and checked. Greatly done each nd every variable functions nd syntax are looks so clean and easy to readable also very specific professionally named. few of things which I guess needs to be changed or corrected like need to check the import modules in some files and missing the parameters with the methods. learned a lot as well.

File: Wordle_One_Word.py [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+attempts = []
+# loop for 6 unique input of strings
+for i in range (1, 7):
+    valid_inp = 0
+    print("Attempt: ", i)
+
+    #Loop till a valid attempt is made
+    while(valid_inp == 0):
+        word_inp = input("Enter your value (Only first 5 characters will be considered): ")
+        word_inp = word_inp[:5]
```
Comment (shivaniraykar) [inline Wordle_One_Word.py:12]: Better to check the word length if it contains 5 letters  or throw error than to use indexes up to 5

File: tipsHelper.py [hunk 13]
```diff
@@ -0,0 +1,72 @@
+from glob import glob
+from unittest import result
+from HW03_Shubham_Dekatey_ui import UiX as ui
+from HW03_Shubham_Dekatey_wordle import WordleX as wordle
```
Comment (shivaniraykar) [inline tipsHelper.py:4]: Throwing me an error: ImportError: cannot import name 'WordleX' from partially initialized module 'HW03_Shubham_Dekatey_wordle' (most likely due to a circular import)
Not sure why!

File: tipsHelper.py [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+        try:
+            data = pd.read_csv("wordRank.csv")
+            csvWordList = data['word'].tolist()
+        except:
+            print("wordRank.csv File Reading error")
+    
+    def filter(goodW, badW):
+        global csvWordList, result
+        
+        # if both the lists are empty, return top 50 from WordRank.csv
```
Comment (shivaniraykar) [inline tipsHelper.py:25]: Nice work! Proper commenting wherever needed!

File: Wordle_One_Word.py (hunk 12 above)
Comment (deka1105) [reply Wordle_One_Word.py:12]: That is handled in a different approach. Appreciate the eye for detail.

File: solver.py [hunk 15]
```diff
@@ -0,0 +1,100 @@
+from glob import glob
+from unittest import result
+from HW03_Shubham_Dekatey_ui import UiX as ui
+from HW03_Shubham_Dekatey_wordle import WordleX as wordle
```
Comment (shivaniraykar) [inline solver.py:4]: You are having circular imports, that's why when we try to run it is throwing an import error.

File: HW03_Shubham_Dekatey_dictionary.py [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+        #     for item in lenWordList:
+        #         f.write("%s\n" % item)
+        try:
+            with open("filterWords.txt", "w") as outfile:
+                outfile.write("\n".join(str(item) for item in lenWordList))
+            log.writeLog("filterWords.txt Created")
+            return len(lenWordList)
+        except:
+            return 0
+            print("filterWords.txt creation error")
```
Comment (shivaniraykar) [inline HW03_Shubham_Dekatey_dictionary.py:28]: print and log statement should have been before your return statement, it wont get executed this way

File: TesterIO.py [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+        """Check if the Frequeny CSV file is created"""
+        import pathlib as pl
+        path = pl.Path("letterFrequency.csv")
+        self.assertEquals((str(path), path.is_file()), (str(path), True))
+        
+    def check_file_rank(self) -> None:
+        """Check if the Rank CSV file is created"""
+        import pathlib as pl
+        path = pl.Path("wordRank.csv")
+        self.assertEquals((str(path), path.is_file()), (str(path), True))
```
Comment (shivaniraykar) [inline TesterIO.py:69]: All the tests are written neatly, you can add more like check if all the words exists in the file

File: HW03_Shubham_Dekatey_wordle.py [hunk 18]
```diff
@@ -0,0 +1,170 @@
+import imp
+from HW03_Shubham_Dekatey_ui import UiX as ui
+from logger import LogX as log
+from solver import SolverX
```
Comment (shivaniraykar) [inline HW03_Shubham_Dekatey_wordle.py:4]: throwing me import errors

File: HW03_Shubham_Dekatey_dictionary.py [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+            log.writeLog("filterWords.txt Created")
+            return len(lenWordList)
+        except:
+            print("filterWords.txt creation error")
+            log.writeLog("filterWords.txt creation error")
+
+    def __str__(self) -> str:
+        return f"Dictionary(filteredList:{str(self.lenWordList)})"
+
+    def filter5letterWord(wordLen):
```
Comment (shivaniraykar) [inline HW03_Shubham_Dekatey_dictionary.py:62]: All the functions are given meaningful names!

**Review by shivaniraykar — COMMENTED** [review]
Overall the code looks great to me. Comments at all the places make the code readable and understandable. All the functions are also given proper names. Just need to recheck the imports as circular imports are used in many places. Also, some minor changes are needed in a few places which are suggested in the comments.

File: HW03_Shubham_Dekatey_dictionary.py [hunk 20]
```diff
@@ -0,0 +1,121 @@
+
+from logger import LogX as log
```
Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_dictionary.py:2]: its throwing me unresolved references..

File: HW03_Shubham_Dekatey_ui.py [hunk 21]
```diff
@@ -0,0 +1,104 @@
+from HW03_Shubham_Dekatey_dictionary import DictionaryX as di
```
Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_ui.py:1]: showing me the error in import. something you should really need to check and fix this modules which is importing.

Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_ui.py:1]: showing me the error in import. something you should really need to check and fix this modules which is importing.

File: HW03_Shubham_Dekatey_dictionary.py [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+        # with open('filterWords.txt', 'w') as f:
+        #     for item in lenWordList:
+        #         f.write("%s\n" % item)
+        try:
+            with open("filterWords.txt", "w") as outfile:
+                outfile.write("\n".join(str(item) for item in lenWordList))
+            log.writeLog("filterWords.txt Created")
+            return len(lenWordList)
+        except:
+            return 0
```
Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_dictionary.py:27]: you can pass the return statement after printing.

Comment (jaiminbd13) [inline HW03_Shubham_Dekatey_dictionary.py:27]: you can pass the return statement after printing.

**Pull request merged**